src/Interpreters/Context.cpp

```diff
@@ -1665,6 +1665,14 @@ zkutil::ZooKeeperPtr Context::getZooKeeper() const
     return shared->zookeeper;
 }
 
+UInt32 Context::getZooKeeperSessionUptime() const
+{
+    std::lock_guard lock(shared->zookeeper_mutex);
+    if (!shared->zookeeper || shared->zookeeper->expired())
+        return 0;
+    return shared->zookeeper->getSessionUptime();
```

Comment:
We told recently that we have to support auxiliary zookeeper for example. But I have another question: 
What will be in case of using `testkeeper` or `keeper`?  In case of using testkeeper, that the session uptime will be equal to server uptime, isn't it? Maybe this is a bit confusing.

Reply:
Session uptime is ZooKeeper client property, it does not depend on ZooKeeper server implementation. It will be equal to ClickHouse server uptime if connection was never lost since ClickHouse server startup (it's always true for testkeeper and may be true sometimes for other implementations).